algoliasearch-java-net/src/main/java/com/algolia/search/JavaNetHttpRequester.java

```diff
@@ -36,12 +36,25 @@
    * @param config HTTPClient agnostic Algolia's configuration.
    */
   public JavaNetHttpRequester(@Nonnull ConfigBase config) {
+    this(
+        config,
+        config.getUseSystemProxy()
+            ? HttpClient.newBuilder().proxy(ProxySelector.getDefault())
+            : HttpClient.newBuilder());
+  }
+
+  /**
+   * Build the reusable instance of httpClient with the given configuration.
+   *
+   * @param config HTTPClient agnostic Algolia's configuration
+   * @param builder Builder for {@linkplain HttpClient HTTP Clients}
+   */
+  public JavaNetHttpRequester(@Nonnull ConfigBase config, HttpClient.Builder builder) {
```

Comment:
IIUC customers encountering the SSL issues will have to instantiate the Java client with a 'custom' builder that's correct as the default values won't be working for them?

Reply:
Yes exactly, since the workaround depend on their Java version and environment.